### PR TITLE
feat: checkout page

### PR DESCRIPTION
**Pull request by pattishin**
## Description

**To test changes locally:**

* Running to see the cart features: `AVOCANO_PURCHASE_MODE=cart npm run start`
   * This should reveal checkout tab and cart, as well as checkout page.
* Running to see the original features: `npm run start`

## Checklist
- [ ] Added steps to reproduce the changes in this pull request
- [ ] Added relevant testing in this pull request
- [ ] Please **merge** this PR for me once it is approved.

### CODE REVIEW

File: client/src/components/checkout-form.js [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    event?.preventDefault();
+    const form = new FormData(this.shadowRoot.querySelector('form') || {});
+    let response;
+
+    if (this.isValidEmail(form.get('email'))) {
+      try {
+        response = await this.onSubmit(form);
+
+        // TODO: getting called to earlier, moving up to parent checkout page to handle success/error handling
+        response?.status?.ok
```
Comment (glasnt) [inline client/src/components/checkout-form.js:66]: I'm also seeing that there is no `response.status.ok`, only a `response.ok`, so suggest: 
```suggestion
        response?.ok
```

In my local test, if I select 'collect', and have a valid email, and don't ask for too many items, I can get a successful cart transaction popup

File: client/src/components/checkout-form.js [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+              @click="${this.toggleFormErrorDialog}"
+            ></mwc-button>
+          </mwc-dialog>`
+        : ''}
+      ${openErrorDialog
+        ? html` <mwc-dialog open>
+            <div class="dialogWrapper">
+              <div>
+                <h2>Oh no! 😭</h2>
+                <div>Unable to complete your checkout.</div>
```
Comment (glasnt) [inline client/src/components/checkout-form.js:129]: I had the thought that we should show the `status` from the returned JSON here, but I can't seem to find that value in the response payload. It's showing up as a 501, so maybe show that status? @grayside for verification on spec

File: client/src/components/checkout-form.js [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+            placeholder="Enter your email"
+            validationMessage="Field is required"
+          ></mwc-textfield>
+          <mwc-select
+            name="type"
+            label="Payment Type"
+            required
+            validationMessage="Field is required"
+          >
+            <mwc-list-item></mwc-list-item>
```
Comment (glasnt) [inline client/src/components/checkout-form.js:104]: if `credit` is the default selected, I don't think we need an empty entry here
```suggestion
```

File: client/src/components/styles/checkout-form.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  }
+
+  .checkoutButton {
+    margin-top: 10px;
+    padding: 10px;
+    color: gray;
+    border-radius: 2px;
+    cursor: pointer;
+    font-family: var(--mdc-typography-font-family, Roboto, sans-serif);
+    border: 1px solid darkgray;
```
Comment (glasnt) [inline client/src/components/styles/checkout-form.js:29]: any reason why this colour doesn't match the `buy` theming in the other purchase mode?

**Comment by glasnt** [comment]
Trying to build with the new flag, I get: 

```

(!) Unused external imports
default imported from external module "path" but never used in "rollup.config.js"
```

File: client/src/utils/fetch.js [hunk 5]
```diff
@@ -28,11 +29,10 @@ export const getProduct = async productId => {
     try {
       const response = await fetch(`${API_URL}/products/${productId}`, {
         method: 'GET',
-        credentials: 'include',
```
Comment (glasnt) [inline client/src/utils/fetch.js:31]: why did these get removed?

File: client/src/components/checkout-form.js (hunk 1 above)
Comment (pattishin) [reply client/src/components/checkout-form.js:66]: Moving this up to parent component! Yup, that was an oversight! 🤦🏼‍♀️ Ty for catching!

File: client/src/components/styles/checkout-form.js (hunk 4 above)
Comment (pattishin) [reply client/src/components/styles/checkout-form.js:29]: No reason! Haven't tightened everything up yet.

Comment (pattishin) [reply client/src/components/styles/checkout-form.js]: Updated color!

**Review by grayside — APPROVED** [review]

**Comment by grayside** [comment]
This looks good to me and a very complete foundation on which to continue iterating. Thank you @pattishin!

**Comment by glasnt** [comment]
CSRF shenanigans greatly helped by https://fractalideas.com/blog/making-react-and-django-play-well-together-single-page-app-model/ !

**Review by grayside — APPROVED** [review]

**Review by glasnt — APPROVED** [review]
🐿️

**Pull request merged**